src/rust/aggregator/src/data/leaderboards.rs

```diff
@@ -0,0 +1,261 @@
+    .await
+    .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
+    sqlx::query!(
+        r#"
+            INSERT INTO competition_indexed_events SELECT txn_version, event_idx, $1 FROM fill_events
+            WHERE NOT EXISTS (
+                SELECT * FROM competition_indexed_events
+                WHERE fill_events.txn_version = competition_indexed_events.txn_version
+                AND fill_events.event_idx = competition_indexed_events.event_idx
+                AND competition_indexed_events.competition_id = $1
+            )
+            AND fill_events.time > $2 AND fill_events.time < $3
+        "#,
+        comp.id,
+        comp.start,
+        comp.end,
+    )
+    .execute(transaction as &mut PgConnection)
+    .await
+    .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
+    sqlx::query!(
+        r#"
+            INSERT INTO competition_indexed_events SELECT txn_version, event_idx, $1 FROM place_limit_order_events
+            WHERE NOT EXISTS (
+                SELECT * FROM competition_indexed_events
+                WHERE place_limit_order_events.txn_version = competition_indexed_events.txn_version
+                AND place_limit_order_events.event_idx = competition_indexed_events.event_idx
+                AND competition_indexed_events.competition_id = $1
+            )
+            AND place_limit_order_events.time > $2 AND place_limit_order_events.time < $3
+        "#,
+        comp.id,
+        comp.start,
+        comp.end,
+    )
+    .execute(transaction as &mut PgConnection)
+    .await
+    .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
+    sqlx::query!(
+        r#"
+            INSERT INTO competition_indexed_events SELECT txn_version, event_idx, $1 FROM place_market_order_events
+            WHERE NOT EXISTS (
+                SELECT * FROM competition_indexed_events
+                WHERE place_market_order_events.txn_version = competition_indexed_events.txn_version
+                AND place_market_order_events.event_idx = competition_indexed_events.event_idx
+                AND competition_indexed_events.competition_id = $1
+            )
+            AND place_market_order_events.time > $2 AND place_market_order_events.time < $3
+        "#,
+        comp.id,
+        comp.start,
+        comp.end,
+    )
+    .execute(transaction as &mut PgConnection)
+    .await
+    .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
+    sqlx::query!(
+        r#"
+            INSERT INTO competition_indexed_events SELECT txn_version, event_idx, $1 FROM place_swap_order_events
+            WHERE NOT EXISTS (
+                SELECT * FROM competition_indexed_events
+                WHERE place_swap_order_events.txn_version = competition_indexed_events.txn_version
+                AND place_swap_order_events.event_idx = competition_indexed_events.event_idx
+                AND competition_indexed_events.competition_id = $1
+            )
```

Comment:
Same here as for all of the above

It's awesome that we can get this all straight from existing data, but it is a bit unreadable

Rather than typing sql directly could we achieve the same result by chaining sqlx methods?